cli/src/commands/runTests.js

```diff
@@ -260,15 +240,24 @@ async function getOrderedFlowBinVersions(): Promise<Array<string>> {
   return _flowBinVersionPromise;
 }
 
+const flowNameRegex = /^flow-v[0-9]+.[0-9]+.[0-9]+$/
+/**
+ * flow filename should be `flow-vx.x.x`
+ * @param {string} name 
+ */
+function checkFlowFilename(name) {
+  return flowNameRegex.test(name);
+}
+
 /**
  * Return the sorted list of cached flow binaries that have previously been retrieved from github
  * and cached in the `.flow-bins-cache` directory.  This function is usually called when a failure
  * has occurred when attempting to refresh the flow releases from github, i.e. offline or over
  * API limit.
  */
 async function getCachedFlowBinVersions(): Promise<Array<string>> {
-  // read the files from the bin dir and remove the leading `flow-v` prefix
-  const versions = (await fs.readdir(path.join(BIN_DIR))).map(dir =>
+  // read the files with name `flow-vx.x.x` from the bin dir and remove the leading `flow-v` prefix
+  const versions = (await fs.readdir(path.join(BIN_DIR))).filter(checkFlowFilename).map(dir =>
```

Comment:
What kind of trash are you trying to filter out? As in, which files are causing issues?

Reply:
If `flow-typed run-tests` crushed BIN_DIR can contain 'garbage files'(such as flow-0.x.x.zip, etc). Without filter `runTests.js` tries check semver and failed. Semver returns undefined.